deepchecks/utils/distribution/preprocessing.py

```diff
@@ -176,16 +179,28 @@ def preprocess_2_cat_cols_to_same_bins(dist1: Union[np.ndarray, pd.Series], dist
         dist1_counter[OTHER_CATEGORY_NAME] = len(dist1) - sum(dist1_counter.values())
         dist2_counter = {k: dist2_counter[k] for k in categories_list}
         dist2_counter[OTHER_CATEGORY_NAME] = len(dist2) - sum(dist2_counter.values())
-        categories_list.append(OTHER_CATEGORY_NAME)
 
     else:
-        dist1_counter = Counter(dist1)
-        dist2_counter = Counter(dist2)
-        categories_list = all_categories
+        dist1_counter = dict(Counter(dist1))
+        dist1_counter[OTHER_CATEGORY_NAME] = 0
+        dist2_counter = dict(Counter(dist2))
+        dist2_counter[OTHER_CATEGORY_NAME] = 0
+
+    for cat in all_categories:
+        if cat not in dist1_counter.keys() or dist1_counter[cat] < len(dist1) * min_category_size_ratio:
+            num_in_dist1 = dist1_counter.pop(cat) if cat in dist1_counter.keys() else 0
+            dist1_counter[OTHER_CATEGORY_NAME] = dist1_counter[OTHER_CATEGORY_NAME] + num_in_dist1
+            num_in_dist2 = dist2_counter.pop(cat) if cat in dist2_counter.keys() else 0
+            dist2_counter[OTHER_CATEGORY_NAME] = dist2_counter[OTHER_CATEGORY_NAME] + num_in_dist2
+
+    if dist1_counter[OTHER_CATEGORY_NAME] == 0 and dist2_counter[OTHER_CATEGORY_NAME] == 0:
+        dist1_counter.pop(OTHER_CATEGORY_NAME)
+        dist2_counter.pop(OTHER_CATEGORY_NAME)
 
     # create an array from counters; this also aligns both counts to the same index
+    categories_list = list(dist1_counter.keys())
     dist1_counts = np.array([dist1_counter[k] for k in categories_list])
-    dist2_counts = np.array([dist2_counter[k] for k in categories_list])
+    dist2_counts = np.array([dist2_counter[k] if k in dist2_counter.keys() else 0 for k in categories_list])
```

Comment:
No need for that. Count objects automatically return 0 if the key does not exist :)

Reply:
BTW, if not, you could use .get(key, 0) instead